### PR TITLE
Avoid duplicating files

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [x] Added a ``news`` entry

`build_steps.sh` is duplicated across azure, travis and drone. This PR puts them in one place to avoid duplicating.

### CODE REVIEW

**Comment by jakirkham** [comment]
Thanks for working on this @isuruf ! 😀

**Comment by mariusvniekerk** [comment]
So this means that we can no longer have provider specific templates in these files?

**Comment by isuruf** [comment]
Yes. They shouldn't in the first place.

**Comment by isuruf** [comment]
If they need platform specific templates, they can be added in a different folder.

**Comment by isuruf** [comment]
@mariusvniekerk, anything else to do?

**Comment by jakirkham** [comment]
Have you already tested this on a feedstock?

**Comment by isuruf** [comment]
Yes, I tried on numpy and it works great. (Found a few issues in the templates that I fixed in conda-smithy)

File: conda_smithy/configure_feedstock.py [hunk 1]
```diff
@@ -750,25 +752,21 @@ def _circle_specific_setup(jinja_env, forge_config, forge_dir, platform):
...
-    if platform == "linux":
-        run_file_name = "run_docker_build"
-    else:
-        run_file_name = "run_osx_build"
-
-    # TODO: Conda has a convenience for accessing nested yaml content.
     template_files = [
-        "{}.sh.tmpl".format(run_file_name),
-        "fast_finish_ci_pr_build.sh.tmpl",
+        ".circleci/fast_finish_ci_pr_build.sh",
```
Comment (scopatz) [inline conda_smithy/configure_feedstock.py]: Should this be `.scripts`?

Comment (isuruf) [reply conda_smithy/configure_feedstock.py]: This is specific to circleci

Comment (scopatz) [reply conda_smithy/configure_feedstock.py]: Ahh OK

File: conda_smithy/feedstock_content/build-locally.py [hunk 2]
```diff
@@ -15,7 +15,7 @@ def setup_environment(ns):
 
 
 def run_docker_build(ns):
-    script = glob.glob(".*/run_docker_build.sh")[0]
+    script = glob.glob(".scripts/run_docker_build.sh")[0]
```
Comment (scopatz) [inline conda_smithy/feedstock_content/build-locally.py]: This doesn't need to be a glob anymore :wink:

**Comment by jakirkham** [comment]
@mariusvniekerk @scopatz, would you have another chance to look at this? 🙂

**Comment by jakirkham** [comment]
AFAICT all comments have already been addressed. Planning on merging end of day Wednesday if no comments.

**Comment by mariusvniekerk** [comment]
Thanks @isuruf

**Comment by isuruf** [comment]
@jakirkham, can you rebase and merge?

**Comment by jakirkham** [comment]
> @jakirkham, can you rebase and merge?

Sorry I'll leave that to you.

**Comment by isuruf** [comment]
This is why I hate formatting PRs.

**Comment by jakirkham** [comment]
Understood. That's why I would like us to start testing formatting in CI so that we don't drift enough where this becomes an issue.

**Comment by jakirkham** [comment]
Probably you can try running the `black` command from CI on this first and then try merging. It's not a science, but that should make things go a bit more smoothly.

**Comment by isuruf** [comment]
> we don't drift enough where this becomes an issue.

It has already become an issue. I'm closing this PR because I don't want to spend any more time on it at the moment

**Comment by jakirkham** [comment]
Sorry the formatting PRs caused you issues.

**Comment by jakirkham** [comment]
Alright I've tried to fix this in PR ( https://github.com/conda-forge/conda-smithy/pull/1171 ).

**Pull request closed without merging**